cpp/src/Slice/Parser.h

```diff
@@ -446,14 +449,13 @@ namespace Slice
         TypeList lookupTypeNoBuiltin(const std::string& identifier, bool emitErrors, bool ignoreUndefined = false);
         ContainedList lookupContained(const std::string& identifier, bool emitErrors);
         ExceptionPtr lookupException(const std::string& identifier, bool emitErrors);
-        [[nodiscard]] UnitPtr unit() const;
```

Comment:
I don't see any reason for the removal of these ``[[nodiscard]]``

Reply:
This wasn't intentional.
I must of missed this when I was resolving the merge conflicts with main.

Reply:
I added these `[[nodiscard]]`s back. And added it to the functions above which should also be marked with it.

These functions aren't marked `const` because they can lead to errors being reported, which causes `errors++;`.
So I'm not surprised the automatic fix didn't mark them `[[nodiscard]]`. But, we should never be called `createStruct` and throwing away the value it returns. I'm sure there's other places that need this fix as well, but I didn't want to get carried away here.